src/buy/components/BuyDropdown.tsx

```diff
@@ -0,0 +1,96 @@
+import { useOnchainKit } from '@/core-react/useOnchainKit';
+import { openPopup } from '@/ui-react/internal/utils/openPopup';
+import { useCallback, useMemo } from 'react';
+import { useAccount } from 'wagmi';
+import { getRoundedAmount } from '../../core/utils/getRoundedAmount';
+import { ONRAMP_BUY_URL } from '../../fund/constants';
+import { getFundingPopupSize } from '../../fund/utils/getFundingPopupSize';
+import { background, cn, color, text } from '../../styles/theme';
+import { ONRAMP_PAYMENT_METHODS } from '../constants';
+import { BuyOnrampItem } from './BuyOnrampItem';
+import { useBuyContext } from './BuyProvider';
+import { BuyTokenItem } from './BuyTokenItem';
+
+export function BuyDropdown() {
+  const { projectId } = useOnchainKit();
+  const { to, fromETH, fromUSDC, from, startPopupMonitor } = useBuyContext();
+  const { address } = useAccount();
+
+  const handleOnrampClick = useCallback(
+    (paymentMethodId: string) => {
+      return () => {
+        const assetSymbol = to?.token?.symbol;
+        let fundAmount = to?.amount;
+        // funding url requires a leading zero if the amount is less than 1
+        if (fundAmount?.[0] === '.') {
+          fundAmount = `0${fundAmount}`;
+        }
+        const fundingUrl = `${ONRAMP_BUY_URL}/one-click?appId=${projectId}&addresses={"${address}":["base"]}&assets=["${assetSymbol}"]&presetCryptoAmount=${fundAmount}&defaultPaymentMethod=${paymentMethodId}`;
```

Comment:
should ["base"] come from onchainkit config?

Reply:
swap only works on base but i will flag this in the docs